src/vec.rs

```diff
@@ -134,6 +134,12 @@ mod tests {
     fn should_panic_if_vec_length_does_not_match_expected() {
         let test_vec = vec![1, 2, 3];
         assert_that(&test_vec).has_length(1);
+    }
+
+    #[test]
+    #[should_panic(expected = "\n\texpected: vec to have length <1>\n\t but was: <3>")]
+    fn should_panic_if_ref_vec_length_does_not_match_expected() {
+        let test_vec = vec![1, 2, 3];
         assert_that(&&test_vec).has_length(1);
```

Comment:
My understanding is that this second failed assertion inside a `#[should_panic]` test wasn't actually being "exercised" because the test would (expectedly) fail as of the first failed assertion

So split these into two separate tests